### ISSUE TITLE
[Bug] `sopa.segmentation.cellpose` falls back to CPU despite `gpu=True` and valid `torch.cuda` context

### ISSUE THREAD

**Issue by rushilp7** [gustaveroussy/sopa#382]
## Description

I am running `sopa` segmentation on a 10x Xenium sample using an HPC cluster with GPUs (LSF scheduler).

I explicitly check `torch.cuda.is_available()` in my script immediately before calling Sopa, and it returns `True`. I calls `sopa.segmentation.cellpose(..., gpu=True)`.

However, the logs indicate that the underlying Cellpose process fails to recognize the GPU, prints `Neither TORCH CUDA nor MPS version not installed/working`, and falls back to CPU (`>>>> using CPU`). This results in extremely slow performance and eventual timeouts/crashes.

I suspect this might be an issue with how `sopa` initializes `cellpose` v4.0.8, or how the GPU context is passed down to the `cpsam` model.

## Reproducing the issue

**1. Environment:**

* `sopa==2.1.11`
* `cellpose==4.0.8` (Note: Using the `cpsam` model by default)
* `torch==2.5.1` (CUDA 12.3 loaded via modules)

**2. Script (`03_segmentation.py`):**

```python
import torch
import sopa
import spatialdata_io as sio

# 1. Verify CUDA availability
print(f"CUDA Available: {torch.cuda.is_available()}") 
# Output in logs is: True

# 2. Load data
sdata = sio.xenium("path/to/data", ...)

# 3. Create patches
sopa.make_image_patches(sdata)

# 4. Run Segmentation
sopa.segmentation.cellpose(
    sdata,
    image_key="morphology_focus",
    model_type="nuclei",
    channels=["DAPI", "DAPI"], 
    diameter=35,
    gpu=True  # Explicitly requesting GPU
)

```

**3. Execution:**
Submitted via LSF requesting 1 GPU (`-gpu "num=1:mode=shared:mps=yes..."`).

## Expected behavior

Sopa should pass the correct GPU device/flags to Cellpose. Cellpose should load the model onto the GPU and process patches using CUDA, resulting in fast segmentation.

## Actual Behavior

The script acknowledges the GPU is present, but Cellpose reverts to CPU.

**Relevant Standard Output (`.out`):**

```text
Host: gpunode04
Queue: gpu1
Method: cellpose
Julia Threads: 36
True  <-- This is the print(torch.cuda.is_available())

```

**Relevant Standard Error (`.err`):**

```text
[INFO] (sopa.segmentation.methods._cellpose) You use cellpose=4.0.8, which requires a pretrained_model. Defaulting to 'cpsam'.
...
[INFO] Running Cellpose (Nuclei)...
...
[INFO] Neither TORCH CUDA nor MPS version not installed/working.
[INFO] >>>> using CPU
[INFO] >>>> using CPU
[INFO] >>>> loading model /home/prushil/.cellpose/models/cpsam

```

## System

* **OS:** Linux (CentOS/RHEL 7 on HPC)
* **Python version:** 3.10
* **Installation method:** pip / conda

<details>
<summary>Dependencies versions (Selected)</summary>

```text
cellpose                   4.0.8
dask                       2024.8.2
sopa                       2.1.11
spatialdata                0.5.0
spatialdata-io             0.3.0
torch                      2.5.1
torchvision                0.20.1
numpy                      2.2.5

```
</details>

<details>
<summary>Dependencies versions (full output of pip list)</summary>

```text
Package                   Version

------------------------- -----------

aiobotocore               3.1.1

aiodns                    3.5.0

aiohappyeyeballs          2.6.1

aiohttp                   3.12.15

aioitertools              0.13.0

aiosignal                 1.4.0

anndata                   0.11.4

annotated-types           0.6.0

anyio                     4.10.0

argon2-cffi               25.1.0

argon2-cffi-bindings      21.2.0

array-api-compat          1.12.0

asciitree                 0.3.3

asttokens                 3.0.0

async-lru                 2.0.5

async-timeout             5.0.1

attrs                     25.4.0

babel                     2.17.0

beautifulsoup4            4.14.2

bleach                    6.3.0

bokeh                     3.8.2

botocore                  1.42.30

Bottleneck                1.4.2

branca                    0.8.1

Brotli                    1.0.9

brotlicffi                1.0.9.2

cellpose                  4.0.8

certifi                   2026.1.4

cffi                      1.17.1

charset-normalizer        3.4.4

click                     8.2.1

cloudpickle               3.1.2

colorcet                  3.1.0

comm                      0.2.3

contourpy                 1.3.1

cycler                    0.12.1

cytoolz                   1.0.1

dask                      2024.8.2

dask-expr                 1.1.13

dask-image                2025.11.0

datashader                0.18.2

debugpy                   1.8.11

decorator                 5.2.1

defusedxml                0.7.1

distributed               2024.8.2

exceptiongroup            1.3.0

executing                 2.2.1

fasteners                 0.20

fastjsonschema            2.21.2

fastremap                 1.17.7

filelock                  3.20.3

fill_voids                2.1.1

flexcache                 0.3

flexparser                0.4

FlowIO                    1.4.0

folium                    0.20.0

fonttools                 4.55.3

frozenlist                1.5.0

fsspec                    2026.1.0

geopandas                 1.1.1

gmpy2                     2.2.1

h11                       0.16.0

h5py                      3.14.0

HeapDict                  1.0.1

html5lib                  1.1

httpcore                  1.0.9

httpx                     0.28.1

idna                      3.11

igraph                    0.11.9

imagecodecs               2024.9.22

ImageIO                   2.37.2

importlib_metadata        8.7.0

ipykernel                 6.31.0

ipython                   8.30.0

jedi                      0.19.2

Jinja2                    3.1.6

jmespath                  1.1.0

joblib                    1.5.3

json5                     0.12.1

jsonschema                4.25.1

jsonschema-specifications 2025.9.1

jupyter_client            8.8.0

jupyter_core              5.9.1

jupyter-events            0.12.0

jupyter-lsp               2.2.5

jupyter_server            2.17.0

jupyter_server_terminals  0.5.3

jupyterlab                4.5.0

jupyterlab_pygments       0.3.0

jupyterlab_server         2.28.0

kiwisolver                1.4.8

lazy_loader               0.4

legacy-api-wrap           1.5

llvmlite                  0.44.0

lmdb                      1.6.2

locket                    1.0.0

loompy                    3.0.8

lxml                      5.3.0

lz4                       4.3.2

mapclassify               2.8.1

markdown-it-py            4.0.0

MarkupSafe                3.0.2

matplotlib                3.10.5

matplotlib-inline         0.2.1

matplotlib-scalebar       0.9.0

mdurl                     0.1.2

mistune                   3.1.2

mkl_fft                   1.3.11

mkl_random                1.2.8

mkl-service               2.4.0

mpi4py                    4.0.3

mpmath                    1.3.0

msgpack                   1.1.1

multidict                 6.1.0

multipledispatch          1.0.0

multiscale_spatial_image  2.0.3

narwhals                  2.7.0

natsort                   8.4.0

nbclient                  0.10.2

nbconvert                 7.16.6

nbformat                  5.10.4

nest-asyncio              1.6.0

networkx                  3.4.2

notebook                  7.5.0

notebook_shim             0.2.4

numba                     0.61.2

numcodecs                 0.13.1

numexpr                   2.11.0

numpy                     2.2.5

numpy-groupies            0.11.3

ome-types                 0.6.3

ome-zarr                  0.11.1

opencv-python             4.13.0.90

opencv-python-headless    4.13.0.90

overrides                 7.7.0

packaging                 25.0

pandas                    2.3.2

pandocfilters             1.5.1

param                     2.3.1

parso                     0.8.5

partd                     1.4.2

patsy                     1.0.2

pexpect                   4.9.0

pillow                    11.3.0

PIMS                      0.7

Pint                      0.24.4

pip                       25.3

platformdirs              4.5.0

pooch                     1.8.2

prometheus_client         0.21.1

prompt_toolkit            3.0.52

propcache                 0.3.1

psutil                    5.9.0

ptyprocess                0.7.0

pure_eval                 0.2.3

pyarrow                   19.0.0

pycares                   4.10.0

pycparser                 2.23

pyct                      0.6.0

pydantic                  2.11.9

pydantic_core             2.33.2

pydantic-extra-types      2.11.0

Pygments                  2.19.2

pynndescent               0.5.13

pyogrio                   0.10.0

pyparsing                 3.2.5

pyproj                    3.6.1

PySocks                   1.7.1

python-dateutil           2.9.0.post0

python-json-logger        4.0.0

pytz                      2025.2

PyYAML                    6.0.2

pyzmq                     26.2.0

readfcs                   2.1.0

referencing               0.37.0

requests                  2.32.5

rfc3339-validator         0.1.4

rfc3986-validator         0.1.1

rich                      14.2.0

roifile                   2025.12.12

rpds-py                   0.22.3

ruff                      0.12.0

s3fs                      2026.1.0

scanpy                    1.11.5

scikit-image              0.25.2

scikit-learn              1.7.1

scipy                     1.15.3

seaborn                   0.13.2

segment-anything          1.0

Send2Trash                1.8.3

session-info2             0.3

setuptools                72.1.0

shapely                   2.1.1

shellingham               1.5.4

six                       1.17.0

slicerator                1.1.0

sniffio                   1.3.1

sopa                      2.1.11

sortedcontainers          2.4.0

soupsieve                 2.5

spatial_image             1.2.3

spatialdata               0.5.0

spatialdata-io            0.3.0

spatialdata-plot          0.2.12

stack_data                0.6.3

statsmodels               0.14.5

sympy                     1.13.1

tblib                     3.2.2

terminado                 0.18.1

texttable                 1.7.0

threadpoolctl             3.5.0

tifffile                  2024.12.12

tinycss2                  1.4.0

toml                      0.10.2

tomli                     2.4.0

toolz                     1.1.0

toposort                  1.10

torch                     2.5.1

torchaudio                2.5.1

torchvision               0.20.1

tornado                   6.5.1

tqdm                      4.67.1

traitlets                 5.14.3

triton                    3.1.0

typer                     0.21.1

typing_extensions         4.15.0

typing-inspection         0.4.2

tzdata                    2025.3

umap-learn                0.5.11

unicodedata2              15.1.0

urllib3                   2.5.0

uvloop                    0.21.0

wcwidth                   0.2.14

webencodings              0.5.1

websocket-client          1.8.0

wheel                     0.46.3

wrapt                     2.0.1

xarray                    2025.4.0

xarray-dataclass          3.0.0

xarray-schema             0.0.3

xarray-spatial            0.5.2

xsdata                    26.1

xyzservices               2025.4.0

yarl                      1.18.0

zarr                      2.18.3

zict                      3.0.0

zipp                      3.23.0

```
</details>

## Questions for the team

1. Is there a known incompatibility regarding GPU handling between `sopa 2.1.11` and `cellpose 4.0+` (specifically the SAM models)?
2. Is there a way to force a specific `device` object in the `sopa.segmentation.cellpose` call rather than just `gpu=True` to help debug if the boolean flag is getting lost?

**Comment by quentinblampey** [comment]
Yes, sorry, I meant `gpu=True` (I was testing this snippet locally, and since I don't have CUDA I passed `gpu=False`...)

> And pretrained_model="nuclei"?

You can try both `"nuclei"` and `"cpsam"`, but it should make no difference in terms of the device being used.
But it's a good thing to try both, just in case. Please let me know which device is used for both `pretrained_model`
